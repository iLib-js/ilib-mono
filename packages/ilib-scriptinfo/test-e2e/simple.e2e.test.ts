/*
 * simple.e2e.test.ts - Simple E2E tests for ilib-scriptinfo package loading
 *
 * Copyright Â© 2025, Box, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import path from "path";
import { execFile } from "child_process";
import { promisify } from "util";

const execFileAsync = promisify(execFile);

interface TestResult {
    stdout: string;
    stderr: string;
    code: number;
}

/**
 * Helper function to run simple test scripts
 */
async function runSimpleTest(testDir: string): Promise<TestResult> {
    const testPath = path.resolve(__dirname, testDir);
    
    let command: string = "pnpm"
    let commandArgs: string[] = ["--silent", "start"]

    try {
        const { stdout, stderr } = await execFileAsync(command, commandArgs, {
            cwd: testPath,
            env: { ...process.env, NODE_ENV: "node" }
        });
        return { stdout, stderr, code: 0 };
    } catch (error: any) {
        return { 
            stdout: error.stdout || "", 
            stderr: error.stderr || "", 
            code: error.code || 1 
        };
    }
}

/**
 * Test suite for simple package loading
 */
describe("ilib-scriptinfo simple package loading", () => {
    
    test("ESM module loading works", async () => {
        const result = await runSimpleTest("esm-test");
        expect(result.code).toBe(0);
        expect(result.stdout).toMatchSnapshot();
        expect(result.stderr).toBe("");
    }, 10000);
    
    test("CommonJS module loading works", async () => {
        const result = await runSimpleTest("legacy-test");
        expect(result.code).toBe(0);
        expect(result.stdout).toMatchSnapshot();
        expect(result.stderr).toBe("");
    }, 10000);
    
    test("TypeScript module loading works", async () => {
        const result = await runSimpleTest("typescript-test");
        expect(result.code).toBe(0);
        expect(result.stdout).toMatchSnapshot();
        expect(result.stderr).toBe("");
    }, 15000);
    
});
