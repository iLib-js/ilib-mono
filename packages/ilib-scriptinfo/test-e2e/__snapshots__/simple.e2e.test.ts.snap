// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ilib-scriptinfo simple package loading CommonJS module loading works 1`] = `
"Lifecycle scripts:
  start
    node index.js

Commands of the root workspace project (to run them, use "pnpm -w run"):
  build
    turbo build
  postinstall
    cp git-hooks/* .git/hooks/
  coverage
    turbo coverage
  coverage:affected
    turbo coverage --affected
  coverage:ci:affected
    turbo coverage --concurrency=5 --affected
  test
    pnpm test:affected
  test:affected
    turbo test --affected --output-logs=new-only
  test:all
    turbo test
  test:all:force
    turbo test --force
  test:ci
    turbo test --concurrency=5
  test:ci:affected
    pnpm test:ci --affected
  test:e2e
    pnpm test:e2e:affected
  test:e2e:affected
    turbo test:e2e --affected --output-logs=new-only
  test:e2e:all
    turbo test:e2e
  test:e2e:all:force
    turbo test:e2e --force
  test:e2e:ci
    turbo test:e2e --concurrency=5
  test:e2e:ci:affected
    turbo test:e2e --concurrency=5 --affected
  doc
    pnpm -F '[origin/main]' doc
  release
    pnpm build && pnpm changeset publish
  run:loctool
    NODE_PATH=packages node --inspect packages/loctool/loctool.js
  run:lint
    NODE_PATH=packages node --inspect packages/ilib-lint/src/index.js
  run:samples
    pnpm -F '{packages/*/samples/*}' run:sample
  run:samples:affected
    pnpm -F '{packages/*/samples/*}[origin/main]' run:sample
"
`;

exports[`ilib-scriptinfo simple package loading ESM module loading works 1`] = `
"Lifecycle scripts:
  start
    node index.js

Commands of the root workspace project (to run them, use "pnpm -w run"):
  build
    turbo build
  postinstall
    cp git-hooks/* .git/hooks/
  coverage
    turbo coverage
  coverage:affected
    turbo coverage --affected
  coverage:ci:affected
    turbo coverage --concurrency=5 --affected
  test
    pnpm test:affected
  test:affected
    turbo test --affected --output-logs=new-only
  test:all
    turbo test
  test:all:force
    turbo test --force
  test:ci
    turbo test --concurrency=5
  test:ci:affected
    pnpm test:ci --affected
  test:e2e
    pnpm test:e2e:affected
  test:e2e:affected
    turbo test:e2e --affected --output-logs=new-only
  test:e2e:all
    turbo test:e2e
  test:e2e:all:force
    turbo test:e2e --force
  test:e2e:ci
    turbo test:e2e --concurrency=5
  test:e2e:ci:affected
    turbo test:e2e --concurrency=5 --affected
  doc
    pnpm -F '[origin/main]' doc
  release
    pnpm build && pnpm changeset publish
  run:loctool
    NODE_PATH=packages node --inspect packages/loctool/loctool.js
  run:lint
    NODE_PATH=packages node --inspect packages/ilib-lint/src/index.js
  run:samples
    pnpm -F '{packages/*/samples/*}' run:sample
  run:samples:affected
    pnpm -F '{packages/*/samples/*}[origin/main]' run:sample
"
`;

exports[`ilib-scriptinfo simple package loading TypeScript module loading works 1`] = `
"Lifecycle scripts:
  start
    tsx index.ts

Commands of the root workspace project (to run them, use "pnpm -w run"):
  build
    turbo build
  postinstall
    cp git-hooks/* .git/hooks/
  coverage
    turbo coverage
  coverage:affected
    turbo coverage --affected
  coverage:ci:affected
    turbo coverage --concurrency=5 --affected
  test
    pnpm test:affected
  test:affected
    turbo test --affected --output-logs=new-only
  test:all
    turbo test
  test:all:force
    turbo test --force
  test:ci
    turbo test --concurrency=5
  test:ci:affected
    pnpm test:ci --affected
  test:e2e
    pnpm test:e2e:affected
  test:e2e:affected
    turbo test:e2e --affected --output-logs=new-only
  test:e2e:all
    turbo test:e2e
  test:e2e:all:force
    turbo test:e2e --force
  test:e2e:ci
    turbo test:e2e --concurrency=5
  test:e2e:ci:affected
    turbo test:e2e --concurrency=5 --affected
  doc
    pnpm -F '[origin/main]' doc
  release
    pnpm build && pnpm changeset publish
  run:loctool
    NODE_PATH=packages node --inspect packages/loctool/loctool.js
  run:lint
    NODE_PATH=packages node --inspect packages/ilib-lint/src/index.js
  run:samples
    pnpm -F '{packages/*/samples/*}' run:sample
  run:samples:affected
    pnpm -F '{packages/*/samples/*}[origin/main]' run:sample
"
`;
